be/src/formats/parquet/stored_column_reader.cpp

```diff
@@ -424,14 +452,43 @@ Status OptionalStoredColumnReader::_read_records_only(size_t* num_records, Colum
     return Status::OK();
 }
 
-Status OptionalStoredColumnReader::_next_page() {
+Status OptionalStoredColumnReader::_next_selected_page(size_t records_to_read, size_t* records_to_skip) {
+    *records_to_skip = 0;
     do {
         RETURN_IF_ERROR(_reader->next_page());
-        _num_values_left_in_cur_page = _reader->num_values();
-    } while (_num_values_left_in_cur_page == 0);
+        size_t num_values = _reader->num_values();
+        if (num_values == 0) {
+            RETURN_IF_ERROR(_reader->skip_page());
+            continue;
+        }
+        if (_select_page(num_values)) {
+            RETURN_IF_ERROR(_reader->load_page());
+            _num_values_left_in_cur_page = num_values;
+            _opts.context->next_row += num_values;
```

Comment:
Do you think we need a method `ColumnReaderContext::fetch_add(int num)` for simplification? I'm afraid sometimes we forget to advance the `ColumnReaderContext::next_row`.